after gcc linking has changed, all the libraries must be explicitely specified 
This patch avoids these linking errors:

Building program /build_disk/poky_build/build0/tmp/work/i586-poky-linux/apt-0.7.14-r4/apt-0.7.14/bin/methods/https
/build_disk/poky_build/build0/tmp/sysroots/x86_64-linux/usr/libexec/i586-poky-linux/gcc/i586-poky-linux/4.5.1/ld: <86>: invalid DSO for symbol `pthread_cancel@@GLIBC_2.0' definition
/build_disk/poky_build/build0/tmp/sysroots/i586-poky-linux/lib/libpthread.so.0: could not read symbols: Bad value
collect2: ld returned 1 exit status
make[2]: *** [/build_disk/poky_build/build0/tmp/work/i586-poky-linux/apt-0.7.14-r4/apt-0.7.14/bin/methods/https] Error 1

Nitin A Kamble <nitin.a.kamble@intel.com>
Date: 2011/01/19

Index: apt-0.7.14/methods/makefile
===================================================================
--- apt-0.7.14.orig/methods/makefile	2008-05-28 06:22:14.000000000 -0700
+++ apt-0.7.14/methods/makefile	2011-01-19 16:30:07.369010214 -0800
@@ -54,7 +54,7 @@
 
 # The https method
 PROGRAM=https
-SLIBS = -lapt-pkg -lcurl
+SLIBS = -lapt-pkg -lcurl -lpthread
 LIB_MAKES = apt-pkg/makefile
 SOURCE = https.cc
 include $(PROGRAM_H)
